www/include/configuration/configObject/host/DB-Func.php

```diff
@@ -2858,3 +2465,124 @@ function applytpl($hosts)
         $hostObj->deployServices($key);
     }
 }
+
+/**
+ * Sanitize all the host parameters from the host form and return a ready to bind array.
+ *
+ * @param array $ret
+ * @return array
+ */
+function sanitizeFormHostParameters(array $ret): array
+{
+    $bindParams = [];
+    foreach ($ret as $inputName => $inputValue) {
+        switch ($inputName) {
+            case 'host_template_model_htm_id':
+            case 'command_command_id':
+            case 'timeperiod_tp_id':
+            case 'timeperiod_tp_id2':
+            case 'command_command_id2':
+            case 'host_max_check_attempts':
+            case 'host_check_interval':
+            case 'host_retry_check_interval':
+            case 'host_freshness_threshold':
+            case 'host_low_flap_threshold':
+            case 'host_high_flap_threshold':
+            case 'host_notification_interval':
+            case 'host_first_notification_delay':
+            case 'host_recovery_notification_delay':
+            case 'host_location':
+            case 'host_acknowledgement_timeout':
+                $bindParams[':' . $inputName] = [
+                    \PDO::PARAM_INT => (filter_var($inputValue, FILTER_VALIDATE_INT) === false)
+                        ? null
+                        : (int) $inputValue
+                ];
+                break;
+            case 'command_command_id_arg1':
+            case 'command_command_id_arg2':
+            case 'host_name':
+            case 'host_alias':
+            case 'host_address':
+            case 'host_snmp_community':
+            case 'host_snmp_version':
+            case 'host_comment':
+            case 'geo_coords':
+                if (!empty($inputValue)) {
+                    $bindParams[':' . $inputName] = [
+                        \PDO::PARAM_STR => (($inputValue = filter_var($inputValue, FILTER_SANITIZE_STRING)) === false)
```

Comment:
SANITIZE_STRING is never false, but could be empty

Reply:
it could be false if you pass false (as a boolean)